lib/puppet/provider/katello_ssl_tool.rb

```diff
@@ -175,15 +175,23 @@ class CertFile < Puppet::Provider
 
     def exists?
       return false unless File.exists?(resource[:path])
-      checksum(expected_content) == checksum(current_content)
+      checksum(expected_content_processed) == checksum(current_content)
```

Comment:
This checksum does an `md5()`. Isn't that a problem in itself for FIPS?

Reply:
Good catch. So apparently puppet by default vendors it's own openssl, so not using the RHEL fips one. As a consequence, the puppet's ruby can still use FIPS non-compliant algorithms (see https://tickets.puppetlabs.com/browse/PUP-7510)

To compare, running on FIPS enabled machine:

```
$ ruby -e 'require "digest/md5"; puts Digest::MD5.digest("eat this")'
md5_dgst.c(82): OpenSSL internal error, assertion failed: Digest MD5 forbidden in FIPS mode!
Aborted
$ /opt/puppetlabs/puppet/bin/ruby -e 'require "digest/md5"; puts Digest::MD5.digest("eat this")'
}T+5��m�$��y�=
```

They have started building a version without bundling openssl for this particular reason (https://github.com/puppetlabs/puppet-agent/commit/7001813f0d8a736f7e8f21a60151ea705f865aff), and in puppet enterprise, there is FIPS listed as separate architecture (https://puppet.com/docs/pe/2017.3/supported_operating_systems.html#puppet-agent-platforms). However, it doesn't look like this builds being available in the community version.

I will open separate PR to replace the md5 here, but it will require additional testing with the puppet's FIPS-enabled version, as their might be dragons hidden in other parts of our installer modules.

Reply:
I'm not sure we actually need to compare checksums. Can't ruby just compare strings? But I'm fine with doing that in a separate patch

Reply:
here you go https://github.com/theforeman/puppet-certs/pull/246